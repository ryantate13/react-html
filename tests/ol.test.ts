import {ol} from '../src';

describe('ol', () => {
    it('renders with children', () => {
        expect(ol({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(ol({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(ol()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(ol({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(ol({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(ol({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(ol({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(ol({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(ol({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(ol({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(ol({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(ol({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(ol({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(ol({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(ol({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(ol({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(ol({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(ol({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(ol({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(ol({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(ol({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(ol({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(ol({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(ol({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(ol({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(ol({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(ol({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(ol({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(ol({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(ol({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(ol({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(ol({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(ol({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(ol({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(ol({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(ol({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(ol({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(ol({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(ol({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(ol({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(ol({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(ol({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(ol({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(ol({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(ol({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(ol({translate: "no"})).toMatchSnapshot();
    });

    it('renders with reversed="true"', () => {
        expect(ol({reversed: true})).toMatchSnapshot();
    });

    it('renders with reversed="false"', () => {
        expect(ol({reversed: false})).toMatchSnapshot();
    });

    it('renders with start="1"', () => {
        expect(ol({start: 1})).toMatchSnapshot();
    });

    it('renders with type="1"', () => {
        expect(ol({type: "1"})).toMatchSnapshot();
    });

    it('renders with type="a"', () => {
        expect(ol({type: "a"})).toMatchSnapshot();
    });

    it('renders with type="A"', () => {
        expect(ol({type: "A"})).toMatchSnapshot();
    });

    it('renders with type="i"', () => {
        expect(ol({type: "i"})).toMatchSnapshot();
    });

    it('renders with type="I"', () => {
        expect(ol({type: "I"})).toMatchSnapshot();
    });
});