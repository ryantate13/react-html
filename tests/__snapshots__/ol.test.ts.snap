// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ol renders arrays of children 1`] = `
<ol>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</ol>
`;

exports[`ol renders with accessKey="a,b,c,d" 1`] = `
<ol
  accessKey="a,b,c,d"
/>
`;

exports[`ol renders with autoCapitalize="characters" 1`] = `
<ol
  autoCapitalize="characters"
/>
`;

exports[`ol renders with autoCapitalize="none" 1`] = `
<ol
  autoCapitalize="none"
/>
`;

exports[`ol renders with autoCapitalize="off" 1`] = `
<ol
  autoCapitalize="off"
/>
`;

exports[`ol renders with autoCapitalize="on" 1`] = `
<ol
  autoCapitalize="on"
/>
`;

exports[`ol renders with autoCapitalize="sentences" 1`] = `
<ol
  autoCapitalize="sentences"
/>
`;

exports[`ol renders with autoCapitalize="words" 1`] = `
<ol
  autoCapitalize="words"
/>
`;

exports[`ol renders with children 1`] = `
<ol>
  TEST
</ol>
`;

exports[`ol renders with className="foo bar baz" 1`] = `
<ol
  className="foo bar baz"
/>
`;

exports[`ol renders with contentEditable="false" 1`] = `
<ol
  contentEditable="false"
/>
`;

exports[`ol renders with contentEditable="true" 1`] = `
<ol
  contentEditable="true"
/>
`;

exports[`ol renders with dir="auto" 1`] = `
<ol
  dir="auto"
/>
`;

exports[`ol renders with dir="ltr" 1`] = `
<ol
  dir="ltr"
/>
`;

exports[`ol renders with dir="rtl" 1`] = `
<ol
  dir="rtl"
/>
`;

exports[`ol renders with draggable="false" 1`] = `
<ol
  draggable="false"
/>
`;

exports[`ol renders with draggable="true" 1`] = `
<ol
  draggable="true"
/>
`;

exports[`ol renders with empty attributes 1`] = `<ol />`;

exports[`ol renders with hidden=false 1`] = `
<ol
  hidden={false}
/>
`;

exports[`ol renders with hidden=true 1`] = `
<ol
  hidden={true}
/>
`;

exports[`ol renders with id="TEST" 1`] = `
<ol
  id="TEST"
/>
`;

exports[`ol renders with inputMode="decimal" 1`] = `
<ol
  inputMode="decimal"
/>
`;

exports[`ol renders with inputMode="email" 1`] = `
<ol
  inputMode="email"
/>
`;

exports[`ol renders with inputMode="none" 1`] = `
<ol
  inputMode="none"
/>
`;

exports[`ol renders with inputMode="numeric" 1`] = `
<ol
  inputMode="numeric"
/>
`;

exports[`ol renders with inputMode="search" 1`] = `
<ol
  inputMode="search"
/>
`;

exports[`ol renders with inputMode="tel" 1`] = `
<ol
  inputMode="tel"
/>
`;

exports[`ol renders with inputMode="text" 1`] = `
<ol
  inputMode="text"
/>
`;

exports[`ol renders with inputMode="url" 1`] = `
<ol
  inputMode="url"
/>
`;

exports[`ol renders with is="foo" 1`] = `
<ol
  is="foo"
/>
`;

exports[`ol renders with itemID="https://google.com/" 1`] = `
<ol
  itemID="https://google.com/"
/>
`;

exports[`ol renders with itemProp="https://google.com/" 1`] = `
<ol
  itemProp="https://google.com/"
/>
`;

exports[`ol renders with itemRef="foo bar" 1`] = `
<ol
  itemRef="foo bar"
/>
`;

exports[`ol renders with itemScope=false 1`] = `
<ol
  itemScope={false}
/>
`;

exports[`ol renders with itemScope=true 1`] = `
<ol
  itemScope={true}
/>
`;

exports[`ol renders with itemType="https://google.com/" 1`] = `
<ol
  itemType="https://google.com/"
/>
`;

exports[`ol renders with lang="" 1`] = `
<ol
  lang=""
/>
`;

exports[`ol renders with lang="es-419" 1`] = `
<ol
  lang="es-419"
/>
`;

exports[`ol renders with reversed=false 1`] = `
<ol
  reversed={false}
/>
`;

exports[`ol renders with reversed=true 1`] = `
<ol
  reversed={true}
/>
`;

exports[`ol renders with slot="TEST" 1`] = `
<ol
  slot="TEST"
/>
`;

exports[`ol renders with spellCheck="false" 1`] = `
<ol
  spellCheck="false"
/>
`;

exports[`ol renders with spellCheck="true" 1`] = `
<ol
  spellCheck="true"
/>
`;

exports[`ol renders with start=1 1`] = `
<ol
  start={1}
/>
`;

exports[`ol renders with style={"color":"#000000"} 1`] = `
<ol
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`ol renders with tabIndex=1 1`] = `
<ol
  tabIndex={1}
/>
`;

exports[`ol renders with title="TEST" 1`] = `
<ol
  title="TEST"
/>
`;

exports[`ol renders with translate="no" 1`] = `
<ol
  translate="no"
/>
`;

exports[`ol renders with translate="yes" 1`] = `
<ol
  translate="yes"
/>
`;

exports[`ol renders with type="1" 1`] = `
<ol
  type="1"
/>
`;

exports[`ol renders with type="A" 1`] = `
<ol
  type="A"
/>
`;

exports[`ol renders with type="I" 1`] = `
<ol
  type="I"
/>
`;

exports[`ol renders with type="a" 1`] = `
<ol
  type="a"
/>
`;

exports[`ol renders with type="i" 1`] = `
<ol
  type="i"
/>
`;

exports[`ol renders without attributes 1`] = `<ol />`;
